### ISSUE TITLE
Website domain lock option breaks the code

### ISSUE THREAD

**Issue by Mrgaton** [MichaelXF/js-confuser#144]
This is my option ![image](https://github.com/user-attachments/assets/829c84c9-96ac-421f-b412-8e1a583adf9e) when i obfuscate the code the result is broken and throws error 
![image](https://github.com/user-attachments/assets/0e120dfe-f2e1-431f-9747-1d430a017256) i know that 2.0 is on alpha so if this will not happend on alpha you can close the issue

**Comment by Mrgaton** [comment]
the pack option is like roadroller? i like it

**Comment by Mrgaton** [comment]
when domain locks fails it just crashes the browser, there is a way of changing that behavior?

**Comment by Mrgaton** [comment]
![image](https://github.com/user-attachments/assets/51a2374c-9456-44ac-a9f3-960c615d727d)

**Comment by Mrgaton** [comment]
![image](https://github.com/user-attachments/assets/917f061d-af17-480f-b25b-74e224197ff7)

**Comment by Mrgaton** [comment]
also if you put in the url a query with the original domain the code would work, why instead you do something more like this

```js
if (!window.location.hostname.endsWith('new--confuser.netlify.app')) {
  console.log('caca');
}
```

**Comment by Mrgaton** [comment]
oh also, the website is pretty cool good job.

**Comment by MichaelXF** [comment]
I appreciate the feedback, the alpha is still be being worked and will replace the current JS-confuser.com.

> also if you put in the url a query with the original domain the code would work, why instead you do something more like this
> 
> ```js
> if (!window.location.hostname.endsWith('new--confuser.netlify.app')) {
>   console.log('caca');
> }
> ```
> 
> also i dont like to repeat that line of code on every function it executes, maybe make it more customisable how many there are and where idk.

This can be added, excluding queries is smart. 

You can try [Custom Locks](https://new--confuser.netlify.app/docs/options/customlocks#custom-locks) for more control over the code inserted and frequency.

**Comment by Mrgaton** [comment]
![image](https://github.com/user-attachments/assets/776382ce-c92c-4815-b9f0-38531c50e0f5)
Ammm this is too overkill

**Comment by Mrgaton** [comment]
Activating everything produces a whoping 1.076.190% increase

**Comment by MichaelXF** [comment]
> ![image](https://private-user-images.githubusercontent.com/68958481/375338564-776382ce-c92c-4815-b9f0-38531c50e0f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExMjU4MDQsIm5iZiI6MTczMTEyNTUwNCwicGF0aCI6Ii82ODk1ODQ4MS8zNzUzMzg1NjQtNzc2MzgyY2UtYzkyYy00ODE1LWI5ZjAtMzg1MzFjNTBlMGY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDA0MTE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxZmQ5N2VhMDdhZGMwZmRjZWUxNTdmYmNkNTYyZTc4YzdlM2UxZmYxOTE5M2E0NzFkY2ZhZGY4N2NiOGY3OWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.08-OnOR8udzqICTyH4RfoB8siIxBI1ZxOh_LjaxLxGU) Ammm this is too overkill
> 
> Activating everything produces a whoping 1.076.190% increase

Yes I agree, better default limits and ability to add custom limits is coming in next alpha release. Also having code-transforms select functions better is being worked on (RGF will apply on outside function, instead of recursively applying to many nested functions)

**Comment by MichaelXF** [comment]
Closing as pre 2.0. Feel free to open again if version 2.0 still has this issue

**Comment by Mrgaton** [comment]
Thank you so much
